core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/leonardo/LeonardoService.scala

```diff
@@ -67,6 +66,14 @@ class LeonardoService(leonardoDAO: LeonardoDAO)(implicit
     pollOperation[ListAppResponse](workspace, ctx, listNonErroredApps)
   }
 
+  def when500OrProcessingException(throwable: Throwable): Boolean =
+    throwable match {
+      case t: WsmApiException     => t.getCode / 100 == 5
+      case t: LeoApiException     => t.getCode / 100 == 5
+      case _: ProcessingException => true
+      case _                      => false
+    }
+
```

Comment:
this method is cut-and-pasted from `org.broadinstitute.dsde.rawls.monitor.workspace.runners.deletion.actions.DeletionAction`, which is now deleted. This is now the only place `when500OrProcessingException` is used so it makes sense to have it here.